sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -112,6 +113,19 @@ package object debug {
       plan foreach {
         case s: WholeStageCodegenExec =>
           codegenSubtrees += s
+        case p: AdaptiveSparkPlanExec =>
+          // Find subtrees from current executed plan of AQE.
+          val executedPlan = p.executedPlan
+          if (executedPlan.find(_.isInstanceOf[WholeStageCodegenExec]).isEmpty) {
```

Comment:
Ah, now I see what you mean. For `isFinalPlan=false`, we may have query stages in the query plan that do not run `queryStageOptimizerRules` yet, and have no `WholeStageCodegenExec`.

I think this is at least consistent with normal EXPLAIN. If the `df` is not run yet, you won't see the leading `*` in plan nodes that indicate whole-stage codegen.

One idea is to trigger AQE during EXPLAIN, so that we can display the final plan. But this won't work for SQL EXPLAIN.

Reply:
We need a bit more thinking about how to do EXPLAIN with AQE.

Reply:
@cloud-fan - I feel `SQL EXPLAIN` is usable, at least for me and several developers in the team. It's kind of annoying that we need to disable AQE every time for this.

Reply:
> One idea is to trigger AQE during EXPLAIN, so that we can display the final plan.

I think users and developers are expecting `EXPLAIN` to be a purely metadata operator, so it might be surprising or inefficient to execute actual query during `EXPLAIN`.

Reply:
After some plan changes (such as join elimination) in AQE, I don't think SQL EXPLAIN is useful as you can never see the final plan. How do you expect SQL EXPLAIN to work for AQE?

Reply:
Because AQE is not changing every query. As an example I gave in unit test case in `ExplainSuite.scala`, for aggregation query, AQE should not affect the behavior for the generated code for aggregation.

Reply:
I am fine with changing the code to be `findSubtrees(aqePlan.executedPlan)`, so the explain code-gen behavior is same as explain, and at least it fixes the case for data frame being executed already (is final plan).

I can just add the change for `SQL EXPLAIN` in our internal fork and discuss in the future. WDYT @cloud-fan ?

Reply:
SGTM. We can also look at how other systems do EXPLAIN with query plan that can change at runtime.

Reply:
@cloud-fan - updated, I agree, will check as well, thanks.